src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -772,6 +776,40 @@ void MergeTreeData::MergingParams::check(const StorageInMemoryMetadata & metadat
             throw Exception("Version column " + version_column + " does not exist in table declaration.", ErrorCodes::NO_SUCH_COLUMN_IN_TABLE);
     };
 
+    /// Check that if the is_deleted column is needed, it exists and is of type UInt8. If exist, version column must be defined too but version checks are not done here.
+    auto check_is_deleted_column = [this, & columns](bool is_optional, const std::string & storage)
+    {
+        if (is_deleted_column.empty())
+        {
+            if (is_optional)
+                return;
+
+            throw Exception("Logical error: is_deleted (" + is_deleted_column + ") column for storage " + storage + " is empty", ErrorCodes::LOGICAL_ERROR);
+        }
+        else
+        {
+            if (version_column.empty() && !is_optional)
+                throw Exception("Logical error: Version column (" + version_column + ") for storage " + storage + " is empty while is_deleted (" + is_deleted_column + ") is not", ErrorCodes::LOGICAL_ERROR);
+
+            bool miss_is_deleted_column = true;
+            for (const auto & column : columns)
+            {
+                if (column.name == is_deleted_column)
+                {
+                    if (!typeid_cast<const DataTypeUInt8 *>(column.type.get()))
+                        throw Exception("is_deleted column (" + is_deleted_column + ") for storage " + storage + " must have type Int8."
+                                " Provided column of type " + column.type->getName() + ".", ErrorCodes::BAD_TYPE_OF_FIELD);
+                    miss_is_deleted_column = false;
+                    break;
+                }
+            }
+
+            if (miss_is_deleted_column)
+                throw Exception("is_deleted column " + is_deleted_column + " does not exist in table declaration.", ErrorCodes::NO_SUCH_COLUMN_IN_TABLE);
```

Comment:
Please add negative tests as well (you can use comments like `-- { serverError NO_SUCH_COLUMN_IN_TABLE }`) to check that a query from an SQL file fails with specific error code)